WordPress/Classes/Networking/ReaderPostServiceRemote.m

```diff
@@ -221,17 +223,27 @@ - (void)fetchPostsFromEndpoint:(NSURL *)endpoint
                            failure:(void (^)(NSError *))failure
 {
     NSString *path = [endpoint absoluteString];
-    
     [self.wordPressComRestApi GET:path
            parameters:params
               success:^(id responseObject, NSHTTPURLResponse *httpResponse) {
                   if (!success) {
                       return;
                   }
+
+                  // NOTE: If an offset param was specified sortRank will be derived
+                  // from the offset + order of the results.
+                  BOOL rankByOffset = [params objectForKey:@"offset"] != nil;
+                  __block CGFloat offset = [[params numberForKey:@"offset"] floatValue];
```

Comment:
> Maybe read and set CGFloat offset = [[params numberForKey:@"offset"] floatValue] in the block within if (rankByOffset), as an optimization.

I don't think that would work as we need to increment the value per pass through the block. Also, having to recreate the var each pass doesn't seem like an optimization as setting it once outside the block.  But maybe I've misunderstood what you had in mind? 

> Not sure if there would be any unexpected results if rankByOffset was evaluated as true while the actual value for [[params numberForKey:@"offset"] floatValue] was 0 as a valid NSNumber obj?

Good eye. I'll add a safety net for that circumstance.